client/components/tinymce/plugins/mentions/mentions.jsx

```diff
@@ -0,0 +1,330 @@
+			this.updatePosition( nextState );
+			this.props.editor.on( 'keydown', this.onKeyDown );
+
+			return;
+		}
+
+		// Visible to invisible state.
+		if ( this.state.showPopover && ! nextState.showPopover ) {
+			this.props.editor.off( 'keydown', this.onKeyDown );
+
+			return;
+		}
+
+		// Update position of popover if cursor has moved to a new line.
+		if ( nextState.showPopover ) {
+			const currentPosition = this.getPosition();
+			const isLineBefore = ( this.top > currentPosition.top ) && ( this.left < currentPosition.left );
+			const isLineAfter = ( this.top < currentPosition.top ) && ( this.left > currentPosition.left );
+
+			if ( isLineBefore || isLineAfter ) {
+				this.updatePosition( nextState, currentPosition );
+			}
+		}
+	}
+
+	componentWillUnmount() {
+		const { editor } = this.props;
+
+		editor.off( 'keydown', this.onKeyDown );
+		editor.off( 'keyup', this.onKeyUp );
+		editor.off( 'click', this.hidePopover );
+	}
+
+	getPosition( { query } = this.state ) {
+		const { editor } = this.props;
+		const range = editor.selection.getRng();
+		const mentionRange = document.createRange();
+
+		// Set range to start at beginning of mention in order to get accurate positioning values.
+		mentionRange.setStart( range.startContainer, range.startOffset - query.length );
+		mentionRange.setEnd( range.endContainer, range.endOffset );
+
+		const rectList = mentionRange.getClientRects();
+
+		if ( rectList.length > 0 ) {
+			const index = rectList.length - 1;
+
+			return {
+				left: rectList[ index ].left,
+				top: rectList[ index ].top,
+			};
+		}
+
+		const nodePosition = tinymce.$( editor.selection.getNode() ).offset();
+
+		return {
+			left: nodePosition.left,
+			top: nodePosition.top,
+		};
+	}
+
+	updatePosition( state, { left, top } = this.getPosition( state ) ) {
+		const { editor, node } = this.props;
+		const mceToolbar = tinymce.$( '.mce-toolbar-grp', editor.getContainer() )[ 0 ];
+		const range = editor.selection.getRng();
+		const rectList = range.getClientRects();
+		let height;
+
+		if ( rectList.length > 0 ) {
+			height = rectList[ 0 ].height;
+		} else {
+			height = editor.selection.getNode().offsetHeight;
+		}
+
+		this.left = left;
+		this.top = top;
+
+		node.style.left = `${ this.left }px`;
+		// 10 is the top position of .mentions__suggestions .popover__inner, which hasn't rendered yet.
+		node.style.top = `${ mceToolbar.offsetHeight + this.top + height - 10 }px`;
+	}
+
+	getQueryText() {
+		// (?:^|\\s) means start of input or whitespace
+		// ([A-Za-z0-9_\+\-]*) means 0 or more instances of: A-Z a-z 0-9 _ + -
+		const matcher = new RegExp( '(?:^|\\s)@([A-Za-z0-9_\+\-]*)$', 'gi' );
+		const range = this.props.editor.selection.getRng();
+		const textBeforeCaret = range.startContainer.textContent.slice( 0, range.startOffset );
+		const match = matcher.exec( textBeforeCaret );
+
+		if ( match && ( match.length > 1 ) ) {
+			return match[ 1 ];
+		}
+
+		return null;
+	}
+
+	getSelectedSuggestionIndex() {
```

Comment:
I think we can use [findIndex](https://lodash.com/docs/4.17.4#findIndex) here. If you pass through undefined or null as the first argument, this method will return -1.

```
findIndex( this.matchingSuggestions, ( suggestion ) => {
    return suggestion.ID === this.state.selectedSuggestionId;
} );
```

I'd also suggest updating the no-match value from null to -1, which is common for these indexOf functions.